src/IDatabaseExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Kros.KORM.Query;
+using Kros.KORM.Query.Sql;
+using System;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace Kros.KORM
+{
+    /// <summary>
+    /// Extensions over <see cref="IDatabase"/>.
+    /// </summary>
+    public static class IDatabaseExtensions
+    {
+        /// <summary>
+        /// Adds the <paramref name="entity"/> to the database.
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type.</typeparam>
+        /// <param name="database"><see cref="IDatabase"/> instance.</param>
+        /// <param name="entity">The entity to add.</param>
+        public static async Task AddAsync<TEntity>(this IDatabase database, TEntity entity) where TEntity : class
```

Comment:
How all this works with transactions?

Reply:
Very well :-) . By default, as with `DbSet`, the call is wrapped in an implicit transaction. Of course, it is possible to explicitly call `database.BeginTransacion()` and wrap multiple calls in one transaction.